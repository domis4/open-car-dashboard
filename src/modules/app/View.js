import React from 'react'
import { Provider } from 'react-redux'
import { store, history } from '../../create-store'
import { compose } from 'recompose'
import { Route, Switch } from 'react-router'
import { ConnectedRouter } from 'connected-react-router'
import { MuiThemeProvider, withStyles, createMuiTheme } from '@material-ui/core/styles'
import Fade from '@material-ui/core/Fade'
import styles from './styles'
import Bootlogo from '../bootlogo'
import Dashboard from '../dashboard'
import MainMenu from '../main-menu'
import MusicPlayer from '../music-player'
import Settings from '../settings'
import Maps from '../google-maps'
import Notifications from '../notifications'

function getTheme(theme) {
  return createMuiTheme({
    overrides: {
      MuiPaper: {
        root: {
          backgroundColor: '#000',
        },
      },
      MuiSvgIcon: {
        root: {
          color: '#fff',
        },
      },
    },
    palette: {
      type: theme.paletteType,
      background: {
        default: theme.paletteType === 'light' ? '#000' : '#fff',
      },
    },
  })
}

const theme = getTheme({
  paletteType: 'dark',
})

const App = ({ classes, isBootIconVisible, setBootIconVisible }) => (
  <Provider {...{ store }}>
    <ConnectedRouter history={history}>
      <MuiThemeProvider theme={theme}>
        <Bootlogo callback={() => setBootIconVisible(false)} />
        {!isBootIconVisible && (
          <Fade {...{ in: !isBootIconVisible, timeout: 600, className: classes.root }}>
            <div className={classes.root}>
              <div className={classes.app}>
                <MainMenu history={history} />
                <main className={classes.content}>
                  <Notifications />
                  <Switch className={classes.router}>
                    <Route exact path="/" component={Dashboard} history={history} />
                    <Route path="/music/:tab?" component={MusicPlayer} history={history} />
                    <Route path="/settings" component={Settings} history={history} />
                    <Route path="/maps" component={Maps} history={history} />
                  </Switch>
                </main>
              </div>
            </div>
          </Fade>
        )}
      </MuiThemeProvider>
    </ConnectedRouter>
  </Provider>
)

export default compose(withStyles(styles))(App)
