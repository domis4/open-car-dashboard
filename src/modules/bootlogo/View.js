import React from 'react'
import Logo from '../../assets/bootlogo/vw.svg'
import SVGInline from 'react-svg-inline'
import Fade from '@material-ui/core/Fade'
import Typography from '@material-ui/core/Typography'

const View = ({ showElement, classes, fadeIn }) => (
  <Fade {...{ in: fadeIn, timeout: 600, className: classes.root }}>
    <div {...{ className: classes.root }}>
      {showElement && (
        <div {...{ className: classes.inner }}>
          <SVGInline {...{ svg: Logo }} />
          <Typography component="h1" className={classes.text}>
            Welcome
          </Typography>
        </div>
      )}
    </div>
  </Fade>
)

export default View
