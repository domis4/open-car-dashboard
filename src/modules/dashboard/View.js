//@flow
import React from 'react'
import Paper from '@material-ui/core/Paper'
import Grid from '@material-ui/core/Grid'
import DashboardApp from '../../components/dashboard-app'
import musicImage from '../music-player/icon.png'
import settingsIcon from '../settings/icon.png'
import googleMapsIcon from '../google-maps/icon.png'

type propTypes = {
  classes: {},
  history: {},
}

const View = ({ classes, history }: propTypes) => (
  <div className={classes.root}>
    <Grid container spacing={24}>
      <Grid item xs={3}>
        <Paper className={classes.paper}>
          <DashboardApp image={musicImage} title="Music" callback={() => history.push('/music')} />
        </Paper>
      </Grid>
      <Grid item xs={3}>
        <Paper className={classes.paper}>
          <DashboardApp image={settingsIcon} title="Settings" callback={() => history.push('/settings')} />
        </Paper>
      </Grid>
      <Grid item xs={3}>
        <Paper className={classes.paper}>
          <DashboardApp image={googleMapsIcon} title="Google Maps" callback={() => history.push('/maps')} />
        </Paper>
      </Grid>
      <Grid item xs={3}>
        <Paper className={classes.paper}>
          <DashboardApp />
        </Paper>
      </Grid>
    </Grid>
  </div>
)

export default View
