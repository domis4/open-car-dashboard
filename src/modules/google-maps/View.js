//@flow
import React from 'react'
import Zoom from '@material-ui/core/Zoom'

type propTypes = {
  classes: {},
  tabValue?: string,
  setTabValue: string => void,
}

const View = ({ classes, tabValue = 'lists', setTabValue }: propTypes) => (
  <Zoom in={true}>
    <iframe className={classes.iframe} src="http://maps.openrouteservice.org" scrolling="no" />
  </Zoom>
)

export default View
