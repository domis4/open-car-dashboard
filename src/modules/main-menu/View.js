import React from 'react'
import Drawer from '@material-ui/core/Drawer'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import Clock from '../../components/clock'
import ArrowBackIos from '@material-ui/icons/ArrowBackIos'
import FilterNone from '@material-ui/icons/FilterNone'

const View = ({ classes, history }) => (
  <Drawer
    variant="permanent"
    classes={{
      paper: classes.drawerPaper,
    }}
  >
    <List className={classes.top}>
      <ListItem button className={classes.center} onClick={() => history.push('/')}>
        <ArrowBackIos />
      </ListItem>
      <ListItem button className={classes.center} onClick={() => history.push('/')}>
        <FilterNone />
      </ListItem>
    </List>
    <List className={classes.bottom}>
      <ListItem className={classes.center}>
        <Clock />
      </ListItem>
      <ListItem button className={classes.center} onClick={() => history.push('/')}>
        <div className={classes.apps} />
      </ListItem>
    </List>
  </Drawer>
)

export default View
