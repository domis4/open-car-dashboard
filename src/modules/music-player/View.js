import React from 'react'
import BottomNavigation from '@material-ui/core/BottomNavigation'
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction'
import ListIcon from '@material-ui/icons/List'
import FavoriteIcon from '@material-ui/icons/Favorite'
import SettingsIcon from '@material-ui/icons/Settings'
import Zoom from '@material-ui/core/Zoom'
import List from './lists'

const View = ({ classes, tabValue = 'lists', setTabValue }) => (
  <Zoom in={true}>
    <div className={classes.root}>
      <List />
      <BottomNavigation value={tabValue} onChange={(_, value) => setTabValue(value)} className={classes.bottom}>
        <BottomNavigationAction label="Lists" value="lists" icon={<ListIcon />} />
        <BottomNavigationAction label="Favorites" value="favorites" icon={<FavoriteIcon />} />
        <BottomNavigationAction label="Settings" value="settings" icon={<SettingsIcon />} />
      </BottomNavigation>
    </div>
  </Zoom>
)

export default View
