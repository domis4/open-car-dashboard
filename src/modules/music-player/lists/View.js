//@flow
import React from 'react'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'

type propTypes = {
  classes: {},
  tabValue?: string,
  setTabValue: string => void,
}

let id = 0
function createData(name, children) {
  id += 1
  return { id, name, children }
}

const rows = [
  createData('Playlist 1'),
  createData('Playlist 2'),
  createData('Playlist 3'),
  createData('Playlist 4'),
  createData('Playlist 5'),
]

const View = ({ classes, tabValue, setTabValue }: propTypes) => (
  <Paper className={classes.root}>
    <Typography component="h1" className={classes.text}>
      Playlists
    </Typography>
    <Table className={classes.table}>
      <TableHead>
        <TableRow>
          <TableCell>Title</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {rows.map(row => {
          return (
            <TableRow key={row.id}>
              <TableCell component="th" scope="row">
                {row.name}
              </TableCell>
            </TableRow>
          )
        })}
      </TableBody>
    </Table>
  </Paper>
)

export default View
