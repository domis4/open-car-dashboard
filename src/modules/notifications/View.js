import React from 'react'
import Snackbar from '../../components/snackbar'

const View = ({ notifications, removeNotification }) => (
  <div>
    {notifications.map(({ uuid, text }) => (
      <Snackbar key={uuid} text={text} close={() => removeNotification(uuid)} />
    ))}
  </div>
)

export default View
