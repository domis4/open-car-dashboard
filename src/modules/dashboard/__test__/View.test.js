import React from 'react'
import { shallow } from 'enzyme'
import View from '../View'

describe('dashboard module View.js', () => {
  it('expects component to match snapshot', () => {
    const wrapper = shallow(
      <View
        {...{
          classes: {},
          history: {},
        }}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })
  it('expects component contain five "Grid" components with styles', () => {
    const wrapper = shallow(
      <View
        {...{
          classes: {},
          history: {},
        }}
      />
    )

    expect(wrapper.find('WithStyles(Grid)')).toHaveLength(5)
  })
  it('expects component contain four "Paper" components with styles', () => {
    const wrapper = shallow(
      <View
        {...{
          classes: {},
          history: {},
        }}
      />
    )

    expect(wrapper.find('WithStyles(Paper)')).toHaveLength(4)
  })
  it('expects component contain four DasbhoardApp "View" components with styles', () => {
    const wrapper = shallow(
      <View
        {...{
          classes: {},
          history: {},
        }}
      />
    )

    expect(wrapper.find('WithStyles(View)')).toHaveLength(4)
  })
})
