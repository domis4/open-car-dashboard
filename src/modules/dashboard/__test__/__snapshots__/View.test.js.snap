// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard module View.js expects component to match snapshot 1`] = `
<div>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      xs={3}
    >
      <WithStyles(Paper)>
        <WithStyles(View)
          callback={[Function]}
          image={
            Object {
              "process": [Function],
            }
          }
          title="Music"
        />
      </WithStyles(Paper)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={3}
    >
      <WithStyles(Paper)>
        <WithStyles(View)
          callback={[Function]}
          image={
            Object {
              "process": [Function],
            }
          }
          title="Settings"
        />
      </WithStyles(Paper)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={3}
    >
      <WithStyles(Paper)>
        <WithStyles(View)
          callback={[Function]}
          image={
            Object {
              "process": [Function],
            }
          }
          title="Google Maps"
        />
      </WithStyles(Paper)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={3}
    >
      <WithStyles(Paper)>
        <WithStyles(View) />
      </WithStyles(Paper)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;
